lib/occupier/mongo/client.rb

```diff
@@ -0,0 +1,108 @@
+module Occupier
+
+  module Mongo
+
+    # @api public
+    class Client
+
+      extend Forwardable
+
+      attr_reader :environment
+
+      def initialize environment = "development", logger = nil
+        @environment = environment.to_s
+        @logger = logger
+        @db_clients = {}
+
+        client
+      end
+
+      def client
+        @client ||= begin
+
+          config = mongo_config[@environment]
+
+          options         =  { logger: @logger }
+          options[:write] =  { w: config['write'].try(:to_i) || 1 }
+          options[:read]  =  { mode: config['read'].try(:to_sym) || :primary }
+
+          addresses = config['hosts'] || ["#{config['host']}:#{config['port']}"]
+
+          ::Mongo::Client.new(addresses, options)
+
+        end
+      end
+
+      def create database_name
+        raise "database already exists" if database_exists? database_name
+        db = db_client(database_name).database
+        db["_dummy"].insert_one({})
+        db
+      end
+
+      def close
+        @db_clients.each { |_name, client| client.close }
+        client.close
+      end
+
+      def database database_name
+        db_client(database_name).database
+      end
+
+      def database! database_name
+        raise "database does not exist" unless database_exists? database_name
+        db_client(database_name).database
+      end
+
+      def reset database_name
+        drop_database database_name
+        create database_name
+        self
+      end
+
+      def database_exists? database_name
+        database_names.one? { |name| name == database_name }
+      end
+
+      def database_names
+        # TODO: test changing this to `client.database_names`
```

Comment:
Missing TODO?

Reply:
@CarlosSoares done, thanks for the heads-up